### PR TITLE
http local_ratelimit: add request_headers_to_add option

### PR DESCRIPTION
**Pull request by williamsfu99**
* Retry of #16076 *

Commit Message:
Introduce the ability to add request headers to forwarded upstream requests when the local HTTP rate limit filter determines (through its configured ruleset) that a request should be rate limited, but is not hard enforcing the incoming traffic.

Additional Description:
This is useful when a backend service should take customized action when a particular traffic volume is reached, rather than forcing request shedding by the proxy.

Risk Level: Low, adding a new feature
Testing: Modified unit tests
Docs Changes: Updated associated doc
Release Notes:
Platform Specific Features:

### CODE REVIEW

File: api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto
Comment (htuch) [inline api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto:69]: This probably needs a different name, maybe `request_headers_to_add_when_not_enforced`?

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
Code LGTM. Waiting for API approval.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Pull request merged**
